Enable new hydra-queue-runner on mimas

mweinelt opened the pull request:
Preparation to deploy the new queue-runner to hydra.nixos.org.

mweinelt committed:
mimas: enable new hydra-queue-runner

mweinelt committed:
mimas: enable new hydra-queue-runner

zowoq commented on non-critical-infra/packages/hydra-queue-runner/default.nix:
```diff
@@ -1,102 +0,0 @@
-{
```
Might want to upstream this as it has the withOtel option and separate packages for the builder and runner.

mweinelt replied:
@Conni2461 Can you pick this?

jfly commented on build/hydra-queue-runner.nix:
```diff
@@ -0,0 +1,92 @@
+      dbUrl = "postgres://hydra@10.0.40.3:5432/hydra";
+      machineFreeFn = "DynamicWithMaxJobLimit";
+      stepSortFn = "WithRdeps";
+      # dispatchTriggerTimerInS?
```
FYI that this comment is here. Vestigial?

mweinelt replied:
Yeah, I was surprised it didn't exist, but forgot the context in which I thought it would.

mweinelt replied:
@Conni2461 Any reason this wasn't exposed in the module?

jfly commented on flake.nix:
```diff
@@ -18,9 +18,10 @@
 
     hydra = {
-      url = "github:NixOS/hydra/a40d42862da88cce78a27dd594e1484a034aac4d";
+      url = "github:NixOS/hydra/token-list-path";
```
Mind adding a comment here saying we can change this to `/master` once https://github.com/NixOS/hydra/pull/1678 has landed?

mweinelt replied:
Honestly, I thought hydra team would review faster than has happened. Will poke the maintainers shortly.

jfly reviewed (approved):
I don't have a deep understanding of this, so take this for the surface review that it is.

All looks reasonable to me, good luck!

mweinelt commented:
Blocked on https://github.com/NixOS/hydra/pull/1686.

mweinelt committed:
wip

Mic92 committed:
monitoring: scrape metrics of new hydra queue runner

The new queue runner only exposes its prometheus metrics on its
localhost REST listener, so since the switch from the old queue runner
its metrics are no longer collected; the existing scrape job still
points at the old runner's now-dead 9198 port. Publish the metrics
through the existing queue-runner.hydra.nixos.org nginx vhost and add a
scrape job for it on pluto.

Mic92 committed:
monitoring: scrape metrics of new hydra queue runner

The new queue runner only exposes its prometheus metrics on its
localhost REST listener, so since the switch from the old queue runner
its metrics are no longer collected; the existing scrape job still
points at the old runner's now-dead 9198 port. Publish the metrics
through the existing queue-runner.hydra.nixos.org nginx vhost and add a
scrape job for it on pluto.

Mic92 committed:
hydra: drop metrics endpoint of old queue runner

The old hydra-queue-runner is disabled and its metrics are now served by
the new queue runner via queue-runner.hydra.nixos.org, so its 9198
listener no longer exists. Remove the stale hydra.conf setting and stop
allowing the port through the firewall.

mweinelt committed:
macs: add hydra-queue-builder as trusted-users to nix

mweinelt committed:
macs: fix queue-builder home dir

mweinelt committed:
builders: align store available threshold with gc settings

mweinelt committed:
builders: require big-parallel for hopeful-rivest and sleepy-brown

mweinelt commented:
We still need https://github.com/NixOS/hydra/issues/1796 for scheduling big-parallel builds well on macs.

mweinelt committed:
macs: kill big-parallel on m1

mweinelt committed:
queue-runner: dedicated nginx access/error log

Mic92 committed:
hydra: stop nginx recycling the queue-runner gRPC connection

Builders keep a single long-lived HTTP/2 channel to the queue runner and
push a large number of small RPCs over it. nginx's default
keepalive_requests (1000) makes nginx send GOAWAY and tear the connection
down mid-stream, cancelling in-flight upload-completion RPCs. These surface
on builders as "connection closed / operation was canceled" and abort the
upload step.

Raise keepalive_requests and keepalive_timeout on the queue-runner vhost so
the connection is not recycled under normal load.

Mic92 committed:
terraform: abort incomplete multipart uploads on cache buckets

The hydra queue runner now uploads large NARs (>4 GiB) to S3 via
multipart uploads with presigned part URLs. When a build aborts
mid-upload the multipart upload is never completed or explicitly
aborted, leaving orphaned parts that accrue storage cost indefinitely.

Add a lifecycle rule to reap incomplete multipart uploads after 7 days
on both the production nix-cache bucket and the shared cache-bucket
module used by the staging caches. Normal uploads finish within an
hour, so 7 days is ample headroom.

Mic92 committed:
hydra-queue-runner: enforce max output size

The Rust queue-runner that replaced the classic hydra-queue-runner does
not read hydra.conf, so its max_output_size was silently unenforced.
Bump hydra to the revision adding the maxOutputSize setting, set it to
4 GiB - 1 B (matching the old prod hydra.conf), and drop the now-dead
max_output_size from hydra.conf.

mweinelt commented on build/hydra-queue-runner.nix:
```diff
@@ -86,6 +86,7 @@ in
       machineFreeFn = "DynamicWithMaxJobLimit";
       stepSortFn = "WithCriticalPath";
       usePresignedUploads = true;
+      maxOutputSize = 4294967295; # 4 GiB - 1 B, matches prod hydra.conf max_output_size
```
when you remove it from hydra.conf, what it matches becomes a dead reference, because the authoritative value is now here.

mweinelt committed:
hydra: drop firewall rule for hydra-notify metrics

The notify service is masked, because we don't use it.
